src/uu/cksum/src/cksum.rs

```diff
@@ -306,6 +311,74 @@ mod options {
     pub const LENGTH: &str = "length";
     pub const RAW: &str = "raw";
     pub const BASE64: &str = "base64";
+    pub const CHECK: &str = "check";
+    pub const TEXT: &str = "text";
+    pub const BINARY: &str = "binary";
+}
+
+/// Determines whether to prompt an asterisk (*) in the output.
+///
+/// This function checks the `tag`, `binary`, and `had_reset` flags and returns a boolean
+/// indicating whether to prompt an asterisk (*) in the output.
+/// It relies on the overrides provided by clap (i.e., `--binary` overrides `--text` and vice versa).
+/// Same for `--tag` and `--untagged`.
+fn prompt_asterisk(tag: bool, binary: bool, had_reset: bool) -> bool {
+    if tag {
+        return false;
+    }
+    if had_reset {
+        return false;
+    }
+    binary
+}
+
+/**
+ * Determine if we had a reset.
+ * This is basically a hack to support the behavior of cksum
+ * when we have the following arguments:
+ * --binary --tag --untagged
+ * Don't do it with clap because if it struggling with the --overrides_with
+ * marking the value as set even if not present
+ */
+fn had_reset(args: &[String]) -> bool {
+    // Indices where "--binary" or "-b", "--tag", and "--untagged" are found
+    let binary_index = args.iter().position(|x| x == "--binary" || x == "-b");
+    let tag_index = args.iter().position(|x| x == "--tag");
+    let untagged_index = args.iter().position(|x| x == "--untagged");
```

Comment:
You seem to actively avoid the `.indices_of().last()` trick that works perfectly fine for `ls`. Is there something that I'm missing?

This code only considers the *first* occurrence of each flag, which leads to no meaningful behavior, because all subsequent flags get ignored. This is demonstrably wrong:

```console
$ cargo run -q --features cksum cksum -a md5 --binary --tag --untagged --binary README.md 
e5773576fc75ff0f8eba14f61587ae28  README.md
$ cksum -a md5 --binary --tag --untagged --binary README.md 
e5773576fc75ff0f8eba14f61587ae28 *README.md
```

Reply:
sure, but I still want to land this PR with this bug.
I don't think it is happening often and I will open a bug and fix it :)